src/QMCWaveFunctions/BsplineFactory/SplineR2R.cpp

```diff
@@ -48,6 +48,113 @@ bool SplineR2R<ST>::write_splines(hdf_archive& h5f)
   return h5f.writeEntry(bigtable, o.str().c_str()); //"spline_0");
 }
 
+/* 
+  This is called by RotatedSPOs::apply_rotation()...
+
+  The implementation here follows that described in [1,2].
+  Each "rotated" orbital, \psi, is composed of a linear combination
+  of SPO's \phi:
+
+  \psi_{i} = \psi_i + \sum_{j} \alpha_{i,j}\phi_{j}
+  
+  where \alpha is an adjustable parameter. This is also close to what 
+  was implemented in the Berkeley branch orbopt scheme. 
+  Cf. MultiSlateDeterminantFast.cpp lines 154 - 170. 
+  
+  Note that this implementation requires that NSPOs > Nelec. In other words,
+  if you want to run a orbopt wfn, you must include some virtual orbitals!
+  
+  Some results (using older Berkeley branch) were published in [3].
+  
+  [1] Filippi & Fahy, JCP 112, (2000)
+  [2] Toulouse & Umrigar, JCP 126, (2007)
+  [3] Townsend et al., PRB 102, (2020)
+*/
+template<typename ST>
+void SplineR2R<ST>::applyRotation(const ValueMatrix& rot_mat, bool use_stored_copy)
```

Comment:
what does use_stored_copy do?  It seems unused but I have a bad feeling about it.

Reply:
No idea, honestly. If I had to hazard a guess I'd say that it's a relic from Berkeley. It seems to be present in the same way for all the orbopt-enabled classes, but it doesn't look to me as if it's actually used anywhere.

Reply:
The `use_stored_copy` variable can be removed with some rearrangement of calls in RotatedSPOs::buildOptVariables. 
At least in the LCAO code, applying orbital rotation modifies the MO coefficient matrix.  When doing optimization, re-applying the rotation needs to start from the original MO coefficients, so there needs to be a copy kept around.  This flag defaults to false, which transfers the original matrix to the copy before multiplying by the rotation matrix.
The function `LCAOrbitalSet::storeParamsBeforeRotation` does the same operation (copy original coefficient matrix to the copy), so the flag can be replaced with appropriate calls to `storeParamsBeforeRotation`.

This was part of some fixes to the LCAO orbital rotation that I never checked in, but I should have.  I can make those changes (like removing use_stored_copy) after this PR gets in.